x-pack/plugins/fleet/server/services/epm/packages/install_state_machine/steps/update_latest_executed_state.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsErrorHelpers } from '@kbn/core/server';
+
+import { PACKAGES_SAVED_OBJECT_TYPE } from '../../../../../constants';
+
+import { auditLoggingService } from '../../../../audit_logging';
+
+import type { InstallContext } from '../_state_machine_package_install';
+
+// Function invoked after each transition
+export const updateLatestExecutedState = async (context: InstallContext) => {
+  const { logger, savedObjectsClient, packageInstallContext, latestExecutedState } = context;
+  const { packageInfo } = packageInstallContext;
+  const { name: pkgName } = packageInfo;
+
+  auditLoggingService.writeCustomSoAuditLog({
+    action: 'update',
+    id: pkgName,
+    savedObjectType: PACKAGES_SAVED_OBJECT_TYPE,
+  });
+  try {
+    return await savedObjectsClient.update(PACKAGES_SAVED_OBJECT_TYPE, pkgName, {
```

Comment:
I wonder if we should add some retry and maybe catch all errors, because errors here might block the state machine.

Reply:
Actually I wrapped the call to the `postTransition` function with a `try...catch `block, so if it fails there it shouldn't block the state machine: https://github.com/elastic/kibana/pull/178657/files#diff-025cd343aee1c99c8f9264f38b3cfa768059d69bb7e138c2ace0ce3c8456d198R89-R96